### DISCUSSION TITLE
Add filter rolloff as additional parameter

### DISCUSSION THREAD

**Discussion by spressi** [bnicenboim/eeguana#224] [Ideas]
Currently, it is possible to set the filter order via the `.config` list. For convenience, it would be nice to be able to set the filter rolloff (in dB/oct) instead (even though this is very easy to do manually - but it convenience always gets more users more quickly). I wrote a small function that does exactly that. Maybe you want to include it in a new version of eeguana.

``` r
rollOffToOrder = function(dB_oct) { #https://www.electronics-tutorials.ws/de/filtern/butterworth-filter-design.html#:~:text=dB/Dekade%20oder-,6n%20dB/Oktave,-.%20Ein%20Filter%20erster
  if (dB_oct %% 6 != 0) warning(paste(dB_oct, "not a multiple of 6. Rounding down filter order. Using", dB_oct %/% 6 * 6, "dB/Oct instead."))
  return(dB_oct %/% 6)
}
```
I currently use it like this: `eeg_filt_band_pass(..., .config=list(method="iir", type="butter", order=rollOffToOrder(24)))` but it could also be a separate parameter of the `.config` list that sets `order` implicitly if `is.na(order)` and `is.na(dB_Oct)==F`

**Comment by bnicenboim** [comment]
looks good! 
Feel free to add it by doing a pull request to the "experimental" branch, just name it `roll_off_order` and add minimal documentation (chatgpt is quite good at it). (Also a unit test won't hurt ;) )

If you have no idea what I'm talking about, open an issue, that's also useful.

**Reply by bnicenboim** [reply]
(Just to be clear, the extra function version, that is easier to maintain and test)
